jsk_pr2_robot/jsk_pr2_startup/jsk_pr2_warning/battery_warning.py

```diff
@@ -50,9 +50,10 @@ def __init__(self):
                 rospy.Duration(self.log_rate), self.log_cb)
 
     def speak(self, sentence):
-        if self.speech_history[sentence] + rospy.Duration(self.warn_repeat_rate) > rospy.Time.now():
+        key = sentence[:4]
```

Comment:
what does `sentence[:4]` means

Reply:
`sentence[:4]` works as a keyword of previously spoken sentence.
The reason why we don't use whole sentence is because it can have variables like (100%, 90%, etc.)

```python
In [28]: speech_history = defaultdict(lambda: rospy.Time(0))

In [29]: speech_history
Out[29]: defaultdict(<function __main__.<lambda>>, {})

In [30]: sentence = "バッテリ温度40度。暑いです。"

In [31]: sentence
Out[31]: '\xe3\x83\x90\xe3\x83\x83\xe3\x83\x86\xe3\x83\xaa\xe6\xb8\xa9\xe5\xba\xa640\xe5\xba\xa6\xe3\x80\x82\xe6\x9a\x91\xe3\x81\x84\xe3\x81\xa7\xe3\x81\x99\xe3\x80\x82'

In [32]: sentence[:4]
Out[32]: '\xe3\x83\x90\xe3'

In [33]: speech_history[sentence[:4]]
Out[33]: rospy.Time[0]

In [34]: speech_history
Out[34]: defaultdict(<function __main__.<lambda>>, {'\xe3\x83\x90\xe3': rospy.Time[0]})
```

But as shown above, python2 does not use Unicode in default (while python3 does), and `sentence[:4]` does not necessarily refer to the first 4 characters in Japanese.
We can use Unicode by just adding `u` to the beginning of string.

```python
In [35]: sentence = u"バッテリ温度40度。暑いです。"

In [36]: sentence
Out[36]: u'\u30d0\u30c3\u30c6\u30ea\u6e29\u5ea640\u5ea6\u3002\u6691\u3044\u3067\u3059\u3002'

In [37]: sentence[:4]
Out[37]: u'\u30d0\u30c3\u30c6\u30ea'
```

cc @takayuki5168